lib/galaxy/model/__init__.py

```diff
@@ -8238,10 +8238,10 @@ def copy_to(self, copied_step, step_mapping, user=None):
                 copied_step.subworkflow = subworkflow
                 copied_subworkflow = subworkflow
             else:
-                # Can this even happen, building a workflow with a subworkflow you don't own ?
-                copied_subworkflow = subworkflow.copy()
+                # Importing a shared workflow with a subworkflow step
+                copied_subworkflow = subworkflow.copy(user=user)
```

Comment:
I think we should make the `user` parameter of this method (`copy_to`) and of `Workflow.copy()` (line 7898) non-optional.